pages/add/add.jsx

```diff
@@ -31,11 +38,117 @@ export default React.createClass({
     event.preventDefault();
     this.setState({numCreatorFields: this.state.numCreatorFields+1});
   },
+  handleFormUpdate(evt, name, field, value) {
+    let formValues = this.state.formValues;
+
+    if ( name === `tags` ) {
+      value = value.split(`#`).map((tag) => {
+        return tag.trim();
+      });
+    }
+
+    formValues[name] = value;
+    this.setState({ formValues });
+  },
+  handleFormSubmit(event) {
+    event.preventDefault();
+    this.refs.basicForm.validates(basicFormIsValid => {
+      this.refs.detailForm.validates(detailFormIsValid => {
+        if (!basicFormIsValid) {
+          console.error(`basic form does not pass validation!`);
+        }
+
+        if (!detailFormIsValid) {
+          console.error(`detail form does not pass validation!`);
+        }
+
+        if (!basicFormIsValid || !detailFormIsValid) {
+          return;
+        }
+
+        this.postEntry(this.state.formValues);
+      });
+    });
+  },
+  getNonce(callback) {
+    Service.nonce()
+      .then((nonce) => {
+        callback(nonce);
+      })
+      .catch((reason) => {
+        this.setState({
+          authError: true
+        });
+        console.error(reason);
+        callback(false);
+      });
+  },
+  postEntry(entryData) {
+    this.getNonce((nonce) => {
+      if (!nonce) { return; }
+
+      let data = Object.assign({}, entryData);
+
+      data.nonce = nonce.nonce;
+      data.csrfmiddlewaretoken = nonce.csrf_token;
+
+      // TODO:FIXME:
+      // Temp fix, "tags", "issues", and "creators" are supposed to be optional fields
+      // but currently Pulse API sees them as required fields
+      // Ticket has been filed https://github.com/mozilla/network-pulse-api/issues/80
+      if (!data.tags) {
+        data.tags = [];
+      }
+
+      if (!data.issues) {
+        data.issues = [];
+      }
+
+      if (!data.creators) {
+        data.creators = [];
+      }
+
+      Service.entries
+        .post(data)
+        .then((response) => {
+          browserHistory.push({
+            pathname: `/entry/${response.id}`
+          });
+        })
+        .catch((reason) => {
+          this.setState({
+            serverError: true
+          });
+          console.error(reason);
+        });
+    });
+  },
   getContentForLoggedInUser() {
+    let authErrorMessage = this.state.authError ? <span className="error">You seem to be logged out at this moment. Try <a href={user.getLoginURL(utility.getCurrentURL())}>logging in</a> again?</span> : null;
```

Comment:
I'm thinking maybe we can temporarily save form values in localstorage so after user is redirected back from login we can restore the form?

(we can file a separate ticket for this)